systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,333 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+        kafkaClientsPodName = kubeClient().listPodsByPrefixInName(kafkaClientsName).get(0).getMetadata().getName();
+
+        LOGGER.info("[TLS] Sending messages to topic {}, cluster {} and message count of {}",
+                topicSourceName, kafkaClusterSourceName, messagesCount);
+        InternalKafkaClient producerInternalKafkaClient = internalKafkaClient.toBuilder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(topicSourceName)
+                .withClusterName(kafkaClusterSourceName)
+                .withKafkaUsername(userTlsSource.getMetadata().getName())
+                .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                .build();
+        producerInternalKafkaClient.checkProducedAndConsumedMessages(producerInternalKafkaClient.sendMessagesTls(), 200);
+
+        LOGGER.info("[TLS] Wait 90s until MM2 refreshes its status.");
+        Thread.sleep(Duration.ofSeconds(90).toMillis());
+
+        LOGGER.info("[TLS] Receiving messages from topic {}, cluster {} and message count of {}",
+                topicTargetName, kafkaClusterTargetName, messagesCount);
+        InternalKafkaClient receiverInternalKafkaClient = producerInternalKafkaClient.toBuilder()
+                .withTopicName(topicTargetName)
+                .withClusterName(kafkaClusterTargetName)
+                .withKafkaUsername(userTlsTarget.getMetadata().getName())
+                .build();
+        receiverInternalKafkaClient.checkProducedAndConsumedMessages(200, receiverInternalKafkaClient.receiveMessagesTls());
+
+        LOGGER.info("[TLS] Going to delete ClientsCA certificates");
+        SecretUtils.deleteSecretWithWait(KafkaResources.clientsCaCertificateSecretName(kafkaClusterSourceName), namespaceName);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName), 3, kafkaSourcePods);
+        DeploymentUtils.waitForDeploymentAndPodsReady(KafkaResources.entityOperatorDeploymentName(kafkaClusterSourceName), 1);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        SecretUtils.deleteSecretWithWait(KafkaResources.clientsCaCertificateSecretName(kafkaClusterTargetName), namespaceName);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(kafkaClusterTargetName), 3, kafkaTargetPods);
+        DeploymentUtils.waitForDeploymentAndPodsReady(KafkaResources.entityOperatorDeploymentName(kafkaClusterTargetName), 1);
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("[TLS] Wait 120s until MM2 refreshes its status.");
+        Thread.sleep(Duration.ofSeconds(240).toMillis());
```

Comment:
In log you have that you are waiting 120s, in code there is 240s. And out of curiosity, why do you need to wait for 4 minutes to mm2 refresh its status?

Reply:
Seems like it has to do something with reconciliation. I would love to know as well, why it takes so long. @scholzj  suggested 90s should be enough. I have tried 60, 120, and lastly 240s did the job. 
Seems like MM2 has to figure out, that topic is already created and load some data of it first.. But that is just my impression...

Reply:
Should not we rather use something like `waitForStatus`?

Reply:
Just out of curiosity, why do you need this wait? For what exactly?